application/src/test/java/org/thingsboard/server/rules/flow/AbstractRuleEngineFlowIntegrationTest.java

```diff
@@ -331,6 +333,15 @@ public void testTwoRuleChainsWithTwoRules() throws Exception {
         RuleChain finalRuleChain = rootRuleChain;
         RuleNode lastRuleNode = secondaryMetaData.getNodes().stream().filter(node -> !node.getId().equals(finalRuleChain.getFirstRuleNodeId())).findFirst().get();
 
+        Awaitility.await().atMost(3, TimeUnit.SECONDS)
```

Comment:
3 sec is prooved to rise a flacky errors. We are generally using a constant like MAX_TIMEOUT with 30 sec. it is anough for anything and never affects a good tests